File: go/datastructure/testlinkedlist_test.go
package datastructure

import (
    "testing"
    lang "github.com/otterop/otterop/go/lang"
    test "github.com/otterop/otterop/go/test"
)


func testLinkedListNew() *TestLinkedList {
    this := new(TestLinkedList)
    this.TestBase = test.TestBaseNew()
    return this
}

type TestLinkedList struct {
    *test.TestBase
}




func (this *TestLinkedList) add()  {
    var a *lang.String = lang.StringWrap(lang.StringLiteral("a"))
    var b *lang.String = lang.StringWrap(lang.StringLiteral("b"))
    var c *lang.String = lang.StringWrap(lang.StringLiteral("c"))
    var d *lang.String = lang.StringWrap(lang.StringLiteral("d"))
    var e *lang.String = lang.StringWrap(lang.StringLiteral("e"))
    var strings *lang.Array[*lang.String] = lang.ArrayNewArray(5, a)
    strings.Set(0, a)
    strings.Set(1, b)
    strings.Set(2, c)
    strings.Set(3, d)
    strings.Set(4, e)
    var expected *lang.Array[*lang.String] = lang.ArrayNewArray(5, a)
    expected.Set(0, e)
    expected.Set(1, d)
    expected.Set(2, a)
    expected.Set(3, b)
    expected.Set(4, c)
    var l *LinkedList[*lang.String] = LinkedListNew[*lang.String]()
    l.AddLast(a)
    l.AddLast(b)
    l.AddLast(c)
    l.AddFirst(d)
    l.AddFirst(e)
    this.AssertTrue(l.Size() == 5, lang.StringWrap(lang.StringLiteral("Size should be 5")))
    var i int = 0
    for it := (l).OOPIterator(); it.HasNext(); {
        s := it.Next()
        this.AssertTrue(s.CompareTo(expected.Get(i)) == 0, lang.StringWrap(lang.StringLiteral("Element mismatch")))
        i++
    }
}

func TestTestLinkedList(t *testing.T) {
    t.Run("add", func(t *testing.T) {
        test := testLinkedListNew()
        test.SetGoTestingT(t)
        test.add()
    })
}

File: ts/datastructure/otterop/datastructure/TestLinkedList.test.ts
import { Array } from '@otterop/lang/Array';
import { String } from '@otterop/lang/String';
import { TestBase } from '@otterop/test/TestBase';
import { LinkedList } from './LinkedList';

export class TestLinkedList extends TestBase {

    public add() : void {
        let a : String = String.wrap("a");
        let b : String = String.wrap("b");
        let c : String = String.wrap("c");
        let d : String = String.wrap("d");
        let e : String = String.wrap("e");
        let strings : Array<String> = Array.newArray(5, a);
        strings.set(0, a);
        strings.set(1, b);
        strings.set(2, c);
        strings.set(3, d);
        strings.set(4, e);
        let expected : Array<String> = Array.newArray(5, a);
        expected.set(0, e);
        expected.set(1, d);
        expected.set(2, a);
        expected.set(3, b);
        expected.set(4, c);
        let l : LinkedList<String> = new LinkedList<String>();
        l.addLast(a);
        l.addLast(b);
        l.addLast(c);
        l.addFirst(d);
        l.addFirst(e);
        this.assertTrue(l.size() == 5, String.wrap("Size should be 5"));
        let i : number = 0;
        for (let s of l) {
            this.assertTrue(s.compareTo(expected.get(i)) == 0, String.wrap("Element mismatch"));
            i++;
        }
    }
}


test('add', () => {
    new TestLinkedList().add();
});
